### PR TITLE
ci : add LoRA test to CI

### LINKED ISSUE #2634: llama : add LoRA test to CI
Add a simple test to [ci/run.sh](https://github.com/ggerganov/llama.cpp/tree/master/ci) to make sure LoRA functionality is OK

### PR DESCRIPTION
**Pull request by slaren**
Downloads a LoRA trained on `shakespeare.txt` and compares the perplexity on this dataset with and without applying the LoRA.

Only for 3B and f16 currently, if it looks ok I can try training another LoRA for 7B, and possibly add tests for quantized models.

Fixes https://github.com/ggerganov/llama.cpp/issues/2634

### CODE REVIEW

**Comment by slaren** [comment]
It would be good to test LoRA with quantized models as well, both with and without a f16 `--lora-base` model, but it looks like we are already very close to the 20 minute time limit. Can we do anything about that?

**Comment by ggerganov** [comment]
I can easily increase it, but the runs will start to take forever eventually as we add more tests.

I can deploy more nodes, so another solution is to group the tests into groups and have different nodes run different groups.

**Comment by slaren** [comment]
I have added a test with q8_0 only for now, hopefully it is not too slow.

**Comment by slaren** [comment]
Looks like it didn't timeout. This should be good enough for now, we can add the rest of the quantized model once we figure the build groups.

Some things to review:
- I am not sure if I am following the naming convention of the files very well
- I noticed that the 7B CUDA perplexity tests have `-t 1`, but not the generation tests, so I added it to these as well

**Comment by ggerganov** [comment]
Let's update this PR after #2398 merge and updating the `convert-lora-to-ggml.py` script to export `.gguf`

**Comment by ggerganov** [comment]
I've bumped the CI timeout to 30 minutes.

For now, we can keep just the `F16` and `Q8_0` LoRAs as I think this covers large portion of the functionality and keeps the time slot small. Will merge this if the CI passes

**Review by ggerganov — APPROVED** [review]

**Pull request merged**
